141617753 total calories meal aw

wlffann opened the pull request:
This PR adds each table's total calories and remaining calories based on the foods in the meal object. It is linked to the date function so that it displays only the calories for the day's meals' foods. 

@case-eee @icorson3 @DavidKnott @jdconrad89

wlffann committed:
Add total calories calculation

[#141617753]

wlffann committed:
Add remaining calories to table

This will calculate and append the remaining calories to each meal
table as the diary page is loaded
[#141617755]

wlffann committed:
Update calorie count's dealing with dates

This PR incorporates David’s work on the date section of the index. It
updates the calories to take in the current date and display the total
calories for that day.
[#141617753]

wlffann committed:
Update table classes and reload total calories

I incorporated the classes that Jason used for the grand totals table
into my html and jQuery. I also updated his function to be called each
time the date changes to update the total to match the day.

[#141617753]

jdconrad89 commented on lib/date.js:
```diff
@@ -4,19 +4,26 @@ $('document').ready(function(){
   currentDate = todayDate;
   add_day();
+  addCaloriesToTable();
+  totalsTable();
```
@wlffann Thanks for adding these for me!

jdconrad89 commented:
@wlffann looks good to me! Love the helper in the tests!

DavidKnott commented on lib/total-calories.js:
```diff
@@ -0,0 +1,48 @@
+
+function calculateMealsRemainingCalories(totalCalories) {
+  return 200 - totalCalories;
+};
```
Good job catching the 200!

DavidKnott commented:
Looks good to merge after the 200 fix.

wlffann committed:
Update remaining calories calculation

I had previously hard coded 200 into the remaining calories calculation
because I am apparently a harsh personal trainer. I updated the
calculation to change based on the meal.
[#141617753]

wlffann committed:
Add styling for remaining

This updates the style of a positive meal remaining to green and a
negative to red.
[#141617759]

DavidKnott commented:
Thanks for making those changes!
